src/esm_environment/esm_environment.py

```diff
@@ -476,7 +469,7 @@ def get_shell_commands(self):
 
         environment = []
         # Write module actions
-        if "module_actions" in self.config:
+        if self.config.get("module_actions", None) is not None:
```

Comment:
`get` by default returns `None` if it is not set.

```suggestion
        if self.config.get("module_actions") is not None:
```